### PR TITLE
Add the ability to ignore a parse result

### PR DESCRIPTION
**Pull request by chrismwendt**
This adds the ability to ignore a parse result to the core DSL. It's useful for ignoring the `\` which precedes escaped characters in string literals (e.g. `"Kevin \"PPMD\" Nanney"`) by obviating the need for a post-processing step that strips the `\`s out of the strings.

This essentially achieves the same result #61, but the approach follows `example/ignore.rb` and the suggestion by @kschiess of making `ignore` part of the DSL.

### CODE REVIEW

File: lib/parslet/atoms/can_flatten.rb [hunk 1]
```diff
@@ -120,7 +120,7 @@ def flatten_repetition(list, named)
       return [] if named && list.empty?
 
       # If there are only strings, concatenate them and return that. 
-      foldl(list) { |s,e| s+e }
+      foldl(list) { |s,e| s.nil? ? e : e.nil? ? s : s + e }
```
Comment (kschiess) [inline lib/parslet/atoms/can_flatten.rb]: Can you special case the `?` by adding two if-statements above this line and leaving it alone? Using `?` cascaded hurts my sensibilities. ;)

**Comment by kschiess** [comment]
Apart from the one small comment, this looks good to me. I'll worry about updating the website past merge. Can you fix the one line and give me a ping? We'll release this just after the merge. 

Good PR, Thank you!

File: lib/parslet/atoms/can_flatten.rb (hunk 1 above)
Comment (chrismwendt) [inline lib/parslet/atoms/can_flatten.rb]: I realized that there's a better solution! Call `compact` on the list: `foldl(list.compact) { |s,e| s+e }`

**Comment by chrismwendt** [comment]
@kschiess: I fixed the cascading `?` issue. Thanks for the review :smiley:

**Pull request merged**